airframe-http/.jvm/src/test/scala/wvlet/airframe/http/HttpRequestMapperTest.scala

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe.http
+import wvlet.airframe.codec.MessageCodecFactory
+import wvlet.airframe.http.HttpMessage.{Request, Response}
+import wvlet.airframe.http.router.{HttpRequestMapper, Route}
+import wvlet.airframe.surface.reflect.ReflectMethodSurface
+import wvlet.airspec.AirSpec
+
+import scala.concurrent.Future
+
+/**
+  *
+ */
+object HttpRequestMapperTest extends AirSpec {
+
+  case class NestedRequest(name: String, msg: String)
+  case class NestedRequest2(id: Int)
+
+  @RPC
+  trait MyApi {
+    def rpc1(p1: String): Unit = {}
```

Comment:
By the way, how does airframe-http work if we define the same name methods like:
```scala
def rpc1(p1: String): Unit = {}
def rpc1(p1: String, p1: Int): Unit = {}
```

Reply:
It will throw a runtime exception when building a Router because these methods will be detected as non-unique mappings. 

I think we can generate different HTTP path mappings for the same name methods. I'll file a ticket.

Reply:
Filed https://github.com/wvlet/airframe/issues/1128